server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -299,6 +307,7 @@ protected BlobStoreRepository(
         this.clusterService = clusterService;
         this.recoverySettings = recoverySettings;
         this.compress = COMPRESS_SETTING.get(metadata.settings());
+        this.supportURLRepo = SUPPORT_URL_REPO.get(metadata.settings());
```

Comment:
I know we discussed this and decided on a cluster setting for this, but does that actually make sense?

I made it a repo setting for now (trivial to change) because I figured:

1. All the other settings we have for the repo are repo level settings.
2. It's weird that this would be the only repo setting that can be dynamically adjusted without re-creating the repo
3. There is the absurdly unlikely chance that someone would actually want to only turn this off for one repo? :P